lib/StripeResource.js

```diff
@@ -192,14 +197,49 @@ StripeResource.prototype = {
       callback.call(
         self,
         new Error.StripeConnectionError({
-          message: 'An error occurred with our connection to Stripe',
+          message: self._generateConnectionErrorMessage(requestRetries),
           detail: error,
         }),
         null
       );
     }
   },
 
+  _shouldRetry: function(res, numRetries) {
+    // Retry on conflict or rate limit error, if we have retries left.
+    if (res && (res.statusCode === 409 || res.statusCode === 429)) {
+      return numRetries < this._stripe.getMaxNetworkRetries();
+    }
+
+    // Retry on connection error, if we have retries left.
+    if (!res && (numRetries < this._stripe.getMaxNetworkRetries())) {
+      return true;
+    }
+
+    return false;
+  },
+
+  _getSleepTime: function(numRetries) {
+    var initialNetworkRetryDelay = this._stripe.getInitialNetworkRetryDelay();
+    var maxNetworkRetryDelay = this._stripe.getMaxNetworkRetryDelay();
+
+    // Apply exponential backoff with initialNetworkRetryDelay on the
+    // number of numRetries so far as inputs. Do not allow the number to exceed
+    // maxNetworkRetryDelay.
+    var sleepSeconds = Math.min(
+      initialNetworkRetryDelay * Math.pow(numRetries - 1, 2),
+      maxNetworkRetryDelay
+    );
+
+    // Apply some jitter.
+    sleepSeconds *= 0.5 * (1 + Math.random());
```

Comment:
this is a range of roughly 0.5-1, correct? Seems too wide a range to me... also odd to have it not centered at a modification of 1... Maybe +/- 10%?

Ah, I see that this is what we did in stripe-php. That's fine; let's just include the range in the comment, as stripe-php does.